### ISSUE TITLE
Spotweb returns all tvshows as movie

### ISSUE THREAD

**Issue by webnar** [spotweb/spotweb#131]
Spotweb returns all tvshows as movie trough the newznab api. At least thats what the clients think. It gets movie and tvshow in the category:

```
<newznab:attr name="category" value="2000"/>
<newznab:attr name="category" value="2040"/>
<newznab:attr name="category" value="5000"/>
```

**Comment by mesa57** [comment]
Please show an api call which delivers wrong results.

**Comment by webnar** [comment]
http://1.1.1.1/spotweb/api?apikey=xxxx&maxage=900&cat=5030%2C5040&limit=100&t=tvsearch&offset=0

**Comment by Mirx** [comment]
Your requesting 5030 & 5040 which are the newznab SD and HD TV categories.

**Comment by webnar** [comment]
yes and the xml provides this:

```
<newznab:attr name="category" value="2000"/>
<newznab:attr name="category" value="2040"/>
<newznab:attr name="category" value="5000"/>
```

**Comment by Mirx** [comment]
Tried to simulate it, see what you mean. In chrome the newznab attributes are showing these vlaues. Actual results are filtered correctly though.
IE11 and Firefox don't the XML that way, so never noticed it before.

**Comment by mesa57** [comment]
The Z cat is not used by all clients, however, if present can be used to identify series (Z1).
Spotnet uses category = 1 (0 in spotweb) with subcats containing b4, d11 or z1 to identify series.
But subcats can also identify ero (d2, d23..26, d72..d75 or z3) or epubs (a5 of z2) . Only if non of those are applicable, it is a movie.

**Comment by Mirx** [comment]
So we agree that taking https://github.com/spotweb/spotweb/blob/3ae47777e2533a4c977c0244ac0664c3f3df84c6/lib/SpotCategories.php#L42L54 into account only a5 should be used to determine a category tag. The rest is quality, and should only be used to enhance a category tag, like 5000 -> 5030.

Wouldn't it make more sense though to use an inverted nabcat2spotcat. As it is available as a more detailed reference then the spotAcat2nabcat provides. Return all the matching cases as XML fields.

As for Z not being available on all clients, that would seriously harm the current API results, because the movies and series section of nabcat2spotcat, and thereby category search results, are based on Zx determination.

P.s. just brainstorming here, as I don't use the XML fields.

**Comment by mesa57** [comment]
It is a minority of very old clients which, I think, are barely used. The Z cat was not part of the spotnet protocol in the beginning.

**Comment by mesa57** [comment]
But coming back to the issue, I do not understand what is exactly wrong.
If you to a tvsearch, it is normal to get back tvshows (series) isn't it ?

**Comment by Mirx** [comment]
The problem appearently is that although the rss/xml results are filtered correctly, all the presented results contain underlying xml fields stating that it is a different category.

Clients like sickbeard and sonarr just rely on the presented titles in the query, and never look at the additional feeds.

I don't know which client webnar is using and why the xml output is generating actual problems.

**Comment by webnar** [comment]
I noticed that sickrage had problems resolving tvshows from spotweb. So i installed nzb hydra the results that came back in nzb hydra for tvshows were movie. nzb hydra looks apparently to the category field in the xml.

**Comment by Mirx** [comment]
I'm tinkering with front-ending the "spotAcat2nabcat" XML generation with a new "spotZcat2nabcat". 

If spotcabz contains Z1, tag as serie (5000)
If spotcabz is another value (Z0, Z2, or Z3) or empty, fall back to AX and BX classification, and tag with current classifications.

That should limit interference with the other classifications, like epub and movie+quality.

Have to take a more detailed look, if I can add a HD/SD classification to Z1 determination, by throwing AX in the mix.

**Comment by mesa57** [comment]
You don't have to code, if you set up the rules clear en unambigiously we will find somebody to code them. So first let's do that.

**Comment by Mirx** [comment]
Because the same sequence is used multiple times in the newznabapi.php, it seems better to alter the functions feeding that sequence than updating all the sequences themselves.

In my mind there are 2 routes:

1) Clear an understandable, but takes some code. I currently stuck why the added if statement isn't giving me the results I want.

```
> # line 425
> 
> 				$nabCat = explode("|", $this->Cat2NewznabCat($spot['category'], $spot['subcata']));
> 				if ($nabCat[0] != "" && is_numeric($nabCat[0])) {
> 					$attr = $doc->createElement('newznab:attr');
> 					$attr->setAttribute('name', 'category');
> 					$attr->setAttribute('value', $nabCat[0]);
> 					$item->appendChild($attr);
> 					
> # line 726
> 
> 	function Cat2NewznabCat($hcat, $cat) {
> 		$result = "-";
> 		$catList = explode("|", $cat);
> 		$cat = $catList[0];
> 		$nr = substr($cat, 1);
> 
> 		# Als $nr niet gevonden kan worden is dat niet erg, het mag echter
> 		# geen Notice veroorzaken.
> 		if (!empty($cat[0])) {
> 			switch ($cat[0]) {
> 				case "a"	: $newznabcat = $this->spotAcat2nabcat(); return @$newznabcat[$hcat][$nr]; break;
> 				case "b"	: $newznabcat = $this->spotBcat2nabcat(); return @$newznabcat[$nr]; break;
> 			} # switch
> 		} # if
> 		
> # line 871
> 	# uiteindelijk de bedoeling als category = 0 en sbcatz = Z1 dan 5xxx. Nu test op werking met algemene formule
> 	# Echter, ondanks dat $spot['subcata'] op regel 425 gebruikt wordt, en dus duidelijk niet leeg is,
> 	# wordt toch de standaard Array gebruikt.
> 	
> 	function spotAcat2nabcat() {
> 		if ( !empty($spot['subcata'])) {
> 			return Array(0 =>
> 				Array(0 => "5000|5030",
> 					  1 => "5000|5030",
> 					  5 => "5000|5030",
> 					  3 => "5000|5030",
> 					  4 => "5000|5040",
> 					  5 => "7000|7050",
> 					  6 => "5000|5040",
> 					  7 => "5000|5040",
> 					  8 => "5000|5040",
> 					  9 => "5000|5040",
> 					  10 => "5000|5030")
> 					 );
> 		} # if
> 		else {
> 			return Array(0 =>
> 				Array(0 => "2000|2030",
> 					  1 => "2000|2030",
> 					  2 => "2000|2030",
> 					  3 => "2000|2030",
> 					  4 => "2000|2040",
> 					  5 => "7000|7020",
> 					  6 => "2000|2040",
> 					  7 => "2000|2040",
> 					  8 => "2000|2040",
> 					  9 => "2000|2040",
> 					  10 => "2000|2030"),
> 			  1 =>
> 				Array(0	=> "3000|3010",
> 					  1 => "3000|3010",
> 					  2 => "3000|3040",
> 					  3 => "3000|3010",
> 					  4 => "3000|3040",
> 					  5 => "3000|3040",
> 					  6 => "3000|3010",
> 					  7 => "3000|3040",
> 					  8 => "3000|3040"),
> 			  2 =>
> 				Array(0 => "4000|4050",
> 					  1 => "4000|4030",
> 					  2 => "TUX",
> 					  3 => "PS",
> 					  4 => "PS2",
> 					  5 => "1000|1020",
> 					  6 => "1000|1040",
> 					  7 => "1000|1050",
> 					  8 => "GBA",
> 					  9 => "GC",
> 					  10 => "1000|1010",
> 					  11 => "1000|1030",
> 					  12 => "1000|1080",
> 					  13 => "4000|4040",
> 					  14 => "4000|4040",
> 					  15 => "4000|4040",
> 					  16 => "3DS"),
> 			  3 =>
> 				Array(0 => "4000|4020",
> 					  1 => "4000|4030",
> 					  2 => "TUX",
> 					  3 => "OS/2",
> 					  4 => "4000|4040",
> 					  5 => "NAV",
> 					  6 => "4000|4040",
> 					  7 => "4000|4040")
> 			);
> 		} # else
> 	} # spotAcat2nabcat
```
2) The efficient way. Haven't started with that one yet, as I can't even get the easy way to work :)

Line 736
				case "a"	: $newznabcat = $this->spotAcat2nabcat(); return @$newznabcat[$hcat][$nr]; break;

The needed alteration would be that if category Z1=1 (series) the "case a" output would regex convert 2xxx (movies) to 5xxx (series)

**Comment by mesa57** [comment]
Do not start to code, but first setup the rules :wink:

**Comment by Mirx** [comment]
Then you have to be a bit clearer on what you mean by "the rules". Cause you lost me there, and don't understand what you are missing in previous info.

Series / TV shouldn't be generating XML child elements "category" with 2xxx, but with 5xxx.

**Comment by mesa57** [comment]
What about a clear translation table between spotnet category's and newznab ones ?

**Comment by Mirx** [comment]
I'll see what I can whip up in an matrix. But most of that is already in the original code.

Looking at TV/movie differentiation only:

Category is in there, but as that does not differentiate between movies and TV, its needs additional qualifiers.
Subcata mapping is in there, but as that is only a quality trigger and does not differentiate between movies and TV, its needs additional qualifiers.
Subcatb mapping is in there, to qualify based on 4 (TV) or 6 (Satellite) and then adds a quality-less 5000 element
Subcatc for subs is in there and working
Subcatd is not used, but it is questionable if you really need it. Lots of wrong tags, and can be gather in a better way. If wanted, should be limited to d11 (television).
Subcatz is the major player missing in the mix, as it is the clear indicator.

**Comment by Mirx** [comment]
Thinking about it a bit more, even the matrix you are requesting is in the code. 

The function nabcat2spotcat has all info that is needed, and like I previously suggested, that should also be used for the xml child category.

Because why generate different tags then the feeds that we are able to request.

**Comment by Mirx** [comment]
@webnar, until this get's a proper fix, you can use this version, which should sort out your problems.

[SpotPage_newznabapi.zip](https://github.com/spotweb/spotweb/files/139792/SpotPage_newznabapi.zip)

Make sure you rename the current file to .old or something to make sure you can revert to the original.

**Comment by mesa57** [comment]
Final fix with commit  	535d8c9
